pandasai.exceptions.PandasAIApiCallError: Training failed because of internal server error try again later!

### System Info

Please share your system info with us.
OS version: ubuntu: 22.04
Python version: python 3.9
The current version of pandasai being used: 2.0.19

### 🐛 Describe the bug

Still 2 days before it is working fine, currently it is throwing error, even update to pandasa==2.0.26 but still same

```
2024-04-01 14:54:36 [INFO] Agent successfully trained on the data
2024-04-01 14:54:36 [INFO] Question: Use 'pd.Timestamp.now()' for relative points in time, e.g. 'this month'. 'this week', 'yesterday', 'last week'. How Many Leads This Week
2024-04-01 14:54:36 [INFO] Running PandasAI with openai LLM...
2024-04-01 14:54:36 [INFO] Prompt ID: e6cc1467-c3dd-40b0-9aec-4876a98add15
2024-04-01 14:54:36 [INFO] Executing Pipeline: GenerateChatPipeline
2024-04-01 14:54:36 [INFO] Executing Step 0: ValidatePipelineInput
2024-04-01 14:54:36 [INFO] Executing Step 1: CacheLookup
2024-04-01 14:54:36 [INFO] Executing Step 2: PromptGeneration
2024-04-01 14:54:37 [ERROR] Pipeline failed on step 2: Training failed because of internal server error try again later!
Traceback (most recent call last):
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/pipelines/chat/generate_chat_pipeline.py", line 281, in run
    output = (self.code_generation_pipeline | self.code_execution_pipeline).run(
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/pipelines/pipeline.py", line 137, in run
    raise e
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/pipelines/pipeline.py", line 101, in run
    step_output = logic.execute(
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/pipelines/chat/prompt_generation.py", line 37, in execute
    self.logger.log(f"Using prompt: {prompt}")
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/prompts/base.py", line 54, in __str__
    return self.to_string()
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/prompts/base.py", line 49, in to_string
    self._resolved_prompt = self.prompt.render(**self.props)
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/prompts/templates/generate_python_code.tmpl", line 19, in top-level template code
    {% include 'shared/vectordb_docs.tmpl' with context %}
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/prompts/templates/shared/vectordb_docs.tmpl", line 1, in top-level template code
    {% if context.vectorstore %}{% set documents = context.vectorstore.get_relevant_qa_documents(context.memory.get_last_message()) %}
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/vectorstores/bamboo_vectorstore.py", line 56, in get_relevant_qa_documents
    docs = self._session.get(
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/helpers/request.py", line 34, in get
    return self.make_request("GET", path, **kwargs)["data"]
  File "/home/purandara/anaconda3/envs/propflo-ai2/lib/python3.9/site-packages/pandasai/helpers/request.py", line 71, in make_request
    raise PandasAIApiCallError(data["message"])
pandasai.exceptions.PandasAIApiCallError: Training failed because of internal server error try again later!
```
@gventuri

--- a/pandasai/vectorstores/bamboo_vectorstore.py
+++ b/pandasai/vectorstores/bamboo_vectorstore.py
@@ -53,10 +53,15 @@ class BambooVectorStore(VectorStore):
         Returns relevant question answers based on search
         """
         k = k or self._max_samples
-        docs = self._session.get(
-            "/training-data/qa/relevant-qa", params={"query": question, "count": k}
-        )
-        return docs["docs"]
+
+        try:
+            docs = self._session.get(
+                "/training-data/qa/relevant-qa", params={"query": question, "count": k}
+            )
+            return docs["docs"]
+        except Exception:
+            self._logger.log("Querying without using training data.")
+            return []
 
     def get_relevant_docs_documents(
         self, question: str, k: Union[int, None] = 3
@@ -67,8 +72,12 @@ class BambooVectorStore(VectorStore):
             question (_type_): list of documents
         """
         k = k or self._max_samples
-
-        docs = self._session.get(
-            "/training-docs/docs/relevant-docs", params={"query": question, "count": k}
-        )
-        return docs["docs"]
+        try:
+            docs = self._session.get(
+                "/training-docs/docs/relevant-docs",
+                params={"query": question, "count": k},
+            )
+            return docs["docs"]
+        except Exception:
+            self._logger.log("Querying without using training docs.")
+            return []

--- a/tests/unit_tests/vectorstores/test_bamboo_vector_store.py
+++ b/tests/unit_tests/vectorstores/test_bamboo_vector_store.py
@@ -70,3 +70,19 @@ class TestBambooVector(unittest.TestCase):
         assert mock_request.call_args_list[0][1] == {
             "params": {"count": 3, "query": "Chroma"}
         }
+
+    @patch("pandasai.helpers.request.Session.make_request", autospec=True)
+    def test_get_qa_raise_exception(self, mock_request):
+        mock_request.side_effect = Exception("Some error occurred")
+        bvs = BambooVectorStore(api_key="dummy_key")
+        docs = bvs.get_relevant_qa_documents("Chroma")
+        mock_request.assert_called_once()
+        assert len(docs) == 0
+
+    @patch("pandasai.helpers.request.Session.make_request", autospec=True)
+    def test_get_docs_raise_exception(self, mock_request):
+        mock_request.side_effect = Exception("Some error occurred")
+        bvs = BambooVectorStore(api_key="dummy_key")
+        docs = bvs.get_relevant_docs_documents("Chroma")
+        mock_request.assert_called_once()
+        assert len(docs) == 0

